File: trab1/ruby/precisao.rb
# encoding: utf-8
include Math

p = 1.0
while p + 1.0 > 1.0 do
  p /= 2.0
end
p *= 2
d = log10(p).floor.abs
n = (log(p)/log(2.0)).abs.floor
puts "Precisao: #{p}"
puts "Núm. digitos significativos decimais: #{d}"
puts "Núm. bits da mantissa: #{n}"

File: trab1/erlang/precisao.erl
-module(precisao).
-export([main/0]).

precisao(1.0, P) -> P;
precisao(S, _) -> precisao((S + 1) / 2, S - 1).

main() ->
    P = precisao(2,0),
    D = abs(trunc(math:log10(P))),
    N = trunc(abs(math:log(P)/math:log(2))),
    io:fwrite("Precisao: ~g~n", [P]),
    io:fwrite("Num. digitos significativos decimais: ~b~n", [D]),
    io:fwrite("Num. bits da mantissa: ~b~n", [N]).

